lib/ansible/modules/database/postgresql/postgresql_pg_hba.py

```diff
@@ -0,0 +1,673 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+'''
+This module is used to manage postgres pg_hba files with Ansible.
+'''
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: postgresql_pg_hba
+short_description: Adds, removes or modifies an rule in a pg_hba file.
+description:
+   - The fundamental function of the module is to create, or delete lines in pg_hba files.
+   - The lines in the file should be in a typical pg_hba form and lines should be unique per key (type, databases, users, source).
+     If they are not unique and the SID is 'the one to change', only one for C(state=present) or none for C(state=absent) of the SID's will remain.
+extends_documentation_fragment: files
+version_added: "2.8"
+options:
+  address:
+    description:
+      - The source address/net where the connections could come from.
+      - Will not be used for entries of I(type)=C(local).
+      - You can also use keywords C(all), C(samehost), and C(samenet).
+    default: "samehost"
+    aliases: [ 'source', 'src' ]
+  backup:
```

Comment:
Maybe change the option backup: yes|no, to backup_file: '/path/to/file', type: 'path' ?

Reply:
I understand what you mean, but this is far more error prone. Do we have access to the path, permissions to write the file, etc. In addition to that, chances are high that they specify the same file for multiple runs, and keep overwriting their backup file.
Therefore I went for letting the module make files where it always can (/tmp) and generating new filenames for every backup.

Reply:
Maybe add the option to set backup_file to a path if they want to hard specify it?

Reply:
yup, the more flexible the better ;) and it's safer (e.g., unexpected reboot).
I think anyawy there will be an issue from people soon with a request to implement a possibility of choosing the bacup file path.

Reply:
Yeah, I agree. Will try to add it. When should we commit this one?

Reply:
Better before 3 March (4 feature freeze).
However if it's possible to add it later without changing ui, that's OK.

Reply:
It is in there.